ecl/resource_ecl_provider_connectivity_tenant_connection_v2.go

```diff
@@ -0,0 +1,461 @@
+}
+
+func resourceProviderConnectivityTenantConnectionV2Delete(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	client, err := config.providerConnectivityV2Client(GetRegion(d, config))
+	if err != nil {
+		return fmt.Errorf("error creating ECL Provider Connectivity client: %s", err)
+	}
+
+	tenantConnection, err := tenant_connections.Get(client, d.Id()).Extract()
+	if err != nil {
+		return CheckDeleted(d, err, "Provider Connectivity Tenant Connection")
+	}
+
+	deviceType := tenantConnection.DeviceType
+
+	if err := tenant_connections.Delete(client, d.Id()).ExtractErr(); err != nil {
+		return fmt.Errorf("error deleting ECL Provider Connectivity Tenant Connection: %s", err)
+	}
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"deleting"},
+		Target:     []string{"DELETED"},
+		Refresh:    waitForTenantConnectionStateDelete(client, d.Id()),
+		Timeout:    d.Timeout(schema.TimeoutDelete),
+		Delay:      5 * time.Second,
+		MinTimeout: 3 * time.Second,
+	}
+
+	if deviceType == "ECL::VirtualNetworkAppliance::VSRX" {
+		stateConf.PollInterval = vnaUpdatePollInterval
+	}
+
+	_, err = stateConf.WaitForState()
+	if err != nil {
+		return fmt.Errorf("Error deleting ECL Tenant Connection: %s", err)
+	}
+
+	d.SetId("")
+
+	return nil
+}
+
+func getServerAttachmentOpts(d *schema.ResourceData) (tenant_connections.Server, error) {
+	var attachmentServers tenant_connections.Server
+	servers := d.Get("attachment_opts_server").([]interface{})
+
+	for _, s := range servers {
+		server := s.(map[string]interface{})
+		segmentationType := server["segmentation_type"].(string)
+		segmentationId := server["segmentation_id"].(int)
+
+		var serverFixedips []tenant_connections.ServerFixedIPs
+
+		fixedIps := server["fixed_ips"].([]interface{})
+		for _, f := range fixedIps {
+			fixedIp := f.(map[string]interface{})
+			ipAddress := fixedIp["ip_address"].(string)
+			subnetId := fixedIp["subnet_id"].(string)
+			serverFixedIp := tenant_connections.ServerFixedIPs{
+				IPAddress: ipAddress,
+				SubnetID:  subnetId,
+			}
+			serverFixedips = append(serverFixedips, serverFixedIp)
+		}
+
+		var serverAaps []tenant_connections.AddressPair
+		aaps := server["allowed_address_pairs"].([]interface{})
+		for _, a := range aaps {
+			aap := a.(map[string]interface{})
+			ipAddress := aap["ip_address"].(string)
+			macAddress := aap["mac_address"].(string)
+			serverAap := tenant_connections.AddressPair{
+				IPAddress:  ipAddress,
+				MACAddress: macAddress,
+			}
+			serverAaps = append(serverAaps, serverAap)
+		}
+
+		attachmentServers = tenant_connections.Server{
+			SegmentationType:    segmentationType,
+			SegmentationID:      segmentationId,
+			FixedIPs:            serverFixedips,
+			AllowedAddressPairs: serverAaps,
+		}
+	}
+
+	log.Printf("[DEBUG] getServerAttachmentOpts: %#v", attachmentServers)
+	return attachmentServers, nil
+}
+
+func getVnaAttachmentOpts(d *schema.ResourceData) (tenant_connections.Vna, error) {
```

Comment:
Please look at the comment for `getServerAttachmentOpts`.

Reply:
Yes, confirm.